lib/utils/extrinsic/extrinsic_test.go

```diff
@@ -0,0 +1,192 @@
+package extrinsic
+
+import (
+	"bytes"
+	"encoding/binary"
+	"reflect"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/optional"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+)
+
+func TestAuthoritiesChangeExt_Encode(t *testing.T) {
+	t.Skip()
+
+	// TODO: scale isn't working for arrays of [32]byte
+	kp, err := sr25519.GenerateKeypair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pub := kp.Public().Encode()
+	pubk := [32]byte{}
+	copy(pubk[:], pub)
+
+	authorities := [][32]byte{pubk}
+
+	ext := NewAuthoritiesChangeExt(authorities)
+
+	enc, err := ext.Encode()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(enc)
+
+	r := &bytes.Buffer{}
+	r.Write(enc)
+	res, err := DecodeExtrinsic(r)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(res, ext) {
+		t.Fatalf("Fail: got %v expected %v", res, ext)
+	}
+}
+
+var alice, _ = common.HexToHash("0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d")
+var bob, _ = common.HexToHash("0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
```

Comment:
can we use the keystore library values here rather than manually setting the hashes?

Reply:
these are from substrate so they dont match ours